README.md

```diff
@@ -24,6 +36,155 @@ External documentation:
+
+### Using Palladio
+
+Palladio adds the two new geometry (SOP) nodes `pldAssign` and `pldGenerate` to houdini. These can be created inside of any geometry object in the `Network Editor`.
+
+#### pldAssign Node
+
+The `pldAssign` node is used for easily attaching the required metadata to the input geometry such that the `pldGenerate` function can then correctly generate a 3D Model.
+When selected, a RPK (rule package), seed and corresponding CGA attributes can directly be set in the parameter editor. 
+When evaluated, the `pldAssign` node attaches the required metadata to the geometry (in the form of primitive attributes) and updates the CGA attributes in the parameter editor accordingly.
+The primitive attributes can also be overwritten before the geometry is passed to a `pldGenerate` node (i.e using an `Attribute Create` node).
+
+#### pldGenerate Node
+The `pldGenerate` node is used to generate the final geometry. 
+It takes an initial shape geometry with metadata as input (i.e. the output of a `pldAssign` node) and executes the CityEngine rule to generate a 3D model.
+In the parameter editor we have the options to:
+* re-emit modified CGA attributes
+* emit material attributes
+* emit CGA reports
+
+### Execute a simple CityEngine Rule
+    
+1. In CityEngine, create the following rule file and share it as a rule package (RPK) to disk somewhere (or download one of the provided examples and share it as a RPK (see [Creating a Rule Package](https://doc.arcgis.com/en/cityengine/latest/help/help-rule-package.htm))):
+   ```
+   attr height = 1
+   
+   @StartRule
+   Init -->
+      extrude(height)
+   ```
+   (Instead you can also directly use [this](doc/data/heightExtrude.rpk) RPK)
+1. In Houdini, in a new scene, add a `grid` node.
+1. Enter the `grid` node, set the `Rows` and `Columns` parameters to two and add the two Palladio nodes `pldAssign` and `pldGenerate`. Connect them like this: ![](doc/img/extrude01.png)
+1. Select the `pldAssign` node and set the `Rule Package` parameter to the path of the previously exported RPK in the parameter editor.
+1. Make the `pldGenerate` node the active render node, this will trigger a "cooking" of the assign and generate nodes and execute the CityEngine Rule. You should now see an extruded grid: ![](doc/img/extrude02.png)
+
+### Overriding Rule Attributes
+
+In the previous section we've used the default value for the `height` attribute. Let's use Houdini tools to modify the rule attribute prior to execution.
+
+#### Using Parameter Editor
+
+1. Make sure the `pldAssign` is evaluated by making the connected `pldGenerate` the active render node
+1. Edit the rule attribute in the parameter editor by changing the value in the generated handle (a rule attribute can be reset by calling `revert to Defaults` or pressing `ctrl + MMB` on the rule name)
+
+![](doc/img/attribute03.png)
```

Comment:
the images in this section are not aligned with the enumeration lists, try to align them